cnf-certification-test/platform/suite.go

```diff
@@ -82,3 +92,153 @@ func testContainersFsDiff(env *provider.TestEnvironment) {
+	var taintedNodes []string
+	var errNodes []string
+
+	// Loop through the debug pods that are tied to each node.
+	for _, dp := range env.DebugPods {
+		// Build a NodeTainted tester object.
+		tester := nodetainted.NewNodeTaintedTester(common.DefaultTimeout, ocpclient.NewOcpClient(), ocpclient.Context{
+			Namespace:     dp.Namespace,
+			Podname:       dp.Name,
+			Containername: dp.Spec.Containers[0].Name,
+		})
+		taintInfo, err := tester.GetKernelTaintInfo()
+		if err != nil {
+			logrus.Error("failed to retrieve kernel taint information from debug pod/host")
+			tnf.ClaimFilePrintf("failed to retrieve kernel taint information from debug pod/host")
+			return
+		}
+		tnf.ClaimFilePrintf(fmt.Sprintf("Pod: %s taintInfo retrieved: %s", dp.Name, taintInfo))
+
+		var message string
+		var taintedBitmap uint64
+		nodeTaintsAccepted := true
+		taintedBitmap, err = strconv.ParseUint(taintInfo, 10, 64) //nolint:gomnd // base 10 and uint64
+
+		if err != nil {
+			logrus.Errorf("failed to parse uint with: %s", err)
+			message = fmt.Sprintf("Could not decode tainted kernel causes (code=%d) for node %s\n", taintedBitmap, dp.Name)
+			tnf.ClaimFilePrintf(message)
+			return
+		}
+		taintMsg, individualTaints := decodeKernelTaints(taintedBitmap)
+
+		// We only will fail the tainted kernel check if the reason for the taint
+		// only pertains to `module was loaded`.
+		logrus.Debug("Checking for 'module was loaded' taints")
+		logrus.Debug("individualTaints", individualTaints)
+		moduleCheck := false
+		for _, it := range individualTaints {
+			if strings.Contains(it, `module was loaded`) {
+				moduleCheck = true
+				break
+			}
+		}
+
+		if moduleCheck {
+			// Retrieve the modules from the node (via the debug pod)
+			modules := tester.GetModulesFromNode(dp.Name)
+			logrus.Debug("Got the modules from node")
+			logrus.Debug("modules", modules)
+			tnf.ClaimFilePrintf("Modules received from the debug pod: ", modules)
+
+			// Loop through the modules looking for `InTree: Y`.
+			// If the module info does not contain this string, the module is "tainted".
+			taintedModules := getOutOfTreeModules(modules, dp.Name, tester)
+			logrus.Debug("Collected all of the tainted modules: ", taintedModules)
+			tnf.ClaimFilePrintf("Kernel Modules loaded that cause taints: ", taintedModules)
+			tnf.ClaimFilePrintf("Modules allowed via configuration: ", env.Config.AcceptedKernelTaints)
+
+			// Looks through the accepted taints listed in the tnf-config file.
+			// If all of the tainted modules show up in the configuration file, don't fail the test.
+			nodeTaintsAccepted = taintsAccepted(env.Config.AcceptedKernelTaints, taintedModules)
+		}
+
+		message = fmt.Sprintf("Decoded tainted kernel causes (code=%d) for node %s : %s\n", taintedBitmap, dp.Name, taintMsg)
+		// Only add the tainted node to the slice if the taint is acceptable.
+		if !nodeTaintsAccepted {
+			taintedNodes = append(taintedNodes, dp.Name)
+		}
+
+		tnf.ClaimFilePrintf(message)
```

Comment:
Do we want to always print in claim file even if the test did pass ?

Reply:
In my opinion, most of the time it is better to keep it as a positive indication (e.g. that it is not just because the string did not print) even if it says zero

Reply:
I moved it so it only prints the message if there's something wrong.